### PR TITLE
Log the hostname the reCAPTCHA was completed on

### PR DESCRIPTION
**Pull request by dbkr**
This could be useful information to have in the logs. Also comment about how & why we don't verify the hostname.

### CODE REVIEW

File: synapse/handlers/auth.py [hunk 1]
```diff
@@ -279,8 +279,17 @@ def _check_recaptcha(self, authdict, clientip):
...
-        if 'success' in resp_body and resp_body['success']:
-            defer.returnValue(True)
+        if 'success' in resp_body:
+            # Note that we do NOT check the hostname here: we explicitly
+            # intend the CAPTCHA to be presented by whatever client the
+            # user is using, we just care that they have completed a CAPTCHA.
+            logger.info(
+                "%s reCAPTCHA from hostname %s",
+                "Successful" if resp_body['success'] else "Failed",
+                resp_body['hostname']
```
Comment (NegativeMjark) [inline synapse/handlers/auth.py]: Maybe use ``.get`` to default the 'hostname' key here?

     resp_body.get('hostname', "<missing hostname key>")

**Comment by NegativeMjark** [comment]
LGTM other than the possibility for KeyErrors

**Comment by dbkr** [comment]
Done. Are these tests still 'expected' to fail?

**Comment by NegativeMjark** [comment]
> Done. Are these tests still 'expected' to fail?

No, but they're now failing due to changes that were snuck in while the CI was broken :(.

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**